lib/pub.dart

```diff
@@ -52,6 +53,8 @@ Future<void> ensurePubspecResolved(
     );
   } on ApplicationException catch (e) {
     throw ResolutionFailedException._(e.toString());
+  } finally {
+    globalHttpClient.close();
```

Comment:
Can we give it a documentation comment.

This is a bit of a hack, I don't mind. But ideally, we'd do something reasonable like creating an a client pass it in, and close it at the end. Or if we don't want to pass it around use zones or something.

Reply:
>  But ideally, we'd do something reasonable like creating an a client pass it in, and close it at the end.
> Or if we don't want to pass it around use zones or something.

Yeah - that would be better...
I'll add a comment.